### PR TITLE
feat(alert-preview): add inbox reason to preview

### PR DESCRIPTION
**Pull request by Zandew**
Attaches inbox reason to each group to display in preview table. No changes needed on frontend.

Screen Shot 2022-12-05 at 10 15 18 AM
^ "New issue"

### CODE REVIEW

File: src/sentry/api/endpoints/project_rule_preview.py [hunk 1]
```diff
@@ -68,5 +69,11 @@ def post(self, request: Request, project) -> Response:
             on_results=lambda x: serialize(x, request.user),
             count_hits=True,
         )
+
+        inbox_details = get_inbox_details([Group(id=int(g["id"])) for g in response.data])
+        for group in response.data:
+            if int(group["id"]) in inbox_details:
```
Comment (rjo100) [inline src/sentry/api/endpoints/project_rule_preview.py:75]: pull out `int(group["id"])` into a separate variable

Comment (rjo100) [inline src/sentry/api/endpoints/project_rule_preview.py:75]: do we need this conditional check? or can we use logic more similar to this:
https://github.com/getsentry/sentry/blob/ad99c1fe7d7378c3761bad757aacf27352c693a6/src/sentry/api/serializers/models/group_stream.py#L277

File: src/sentry/api/endpoints/project_rule_preview.py [hunk 2]
```diff
@@ -68,5 +69,10 @@ def post(self, request: Request, project) -> Response:
             on_results=lambda x: serialize(x, request.user),
             count_hits=True,
         )
+
+        inbox_details = get_inbox_details([Group(id=int(g["id"])) for g in response.data])
```
Comment (rjo100) [inline src/sentry/api/endpoints/project_rule_preview.py:73]: https://github.com/getsentry/sentry/blob/ece6d6c751166e1fe05d6b187f207179f1e8cd8f/src/sentry/models/groupinbox.py#L126

it looks like `get_inbox_details` just pulls the ids out of the `Group` object anyways. assuming there's no consistency issues, can we just make `get_inbox_details` a wrapper for a method that takes in a list of `id`? that way we don't have to hit the db here

Comment (Zandew) [reply src/sentry/api/endpoints/project_rule_preview.py:73]: sounds good

File: src/sentry/api/endpoints/project_rule_preview.py [hunk 3]
```diff
@@ -68,5 +69,10 @@ def post(self, request: Request, project) -> Response:
             on_results=lambda x: serialize(x, request.user),
             count_hits=True,
         )
+
+        inbox_details = get_inbox_details_from_ids([int(g["id"]) for g in response.data])
+        for group in response.data:
+            group["inbox"] = inbox_details.get(int(group["id"]))
```
[the remark this answers is not in the document]
Comment (Zandew) [reply src/sentry/api/endpoints/project_rule_preview.py:75]: thanks! good idea

**Review by rjo100 — APPROVED** [review]

**Pull request merged**